llarp/dht/messages/gotrouter.hpp

```diff
@@ -18,8 +18,9 @@ namespace llarp
       {
       }
       GotRouterMessage(const Key_t& from, uint64_t id,
-                       std::vector< RouterContact > results, bool tunneled)
-          : IMessage(from), R(std::move(results)), txid(id), relayed(tunneled)
+                       const std::vector< RouterContact >& results,
+                       bool tunneled)
+          : IMessage(from), R(results), txid(id), relayed(tunneled)
       {
```

Comment:
I really think these are effectively equivalent in terms of safety. The former version is better, however, because the caller could `std::move()` their `std::vector<RouterContact>` which would avoid any copy at all. The latter version requires a minimum one copy in all cases.

Reply:
Yeah, the original guarantees at most one copy, the new one can have two.